go/arrow/scalar/nested.go

```diff
@@ -744,3 +745,72 @@ func (s *DenseUnion) ChildValue() Scalar { return s.Value }
 func NewDenseUnionScalar(v Scalar, code arrow.UnionTypeCode, dt *arrow.DenseUnionType) *DenseUnion {
 	return &DenseUnion{scalar: scalar{dt, v.IsValid()}, TypeCode: code, Value: v}
 }
+
+type RunEndEncoded struct {
+	scalar
+
+	RunLength int64
```

Comment:
Nitpick: I called it `run_end` in C++ and it fit more naturally with the rest of the code.

For example:
 - it's typed by the "run end" type
 - the value 0 is invalid (no run ends array should contain 0) representing the null scalar